File: frontend/src/types/Job.ts
export type CodeTaskType = "write-code" | "fix-code" | "clean-code";
export type DataTaskType = "analyze-data";
export type PMTaskType   = "review-comments" | "review-document" | "ux-problem" | "stakeholder-notes";

export type TaskType = CodeTaskType | DataTaskType | PMTaskType;

export interface Job {
  id: string;
  icon: string;
  titleAr: string;
  titleEn: string;
  descriptionPrimary: string;
  descriptionSecondary: string;
  skills: string[];
  tasks: Task[];
}

export interface Task {
  id: string;
  type: TaskType;
  title: string;
  duration: string;
  fullTitle: string;
  timeRange: string;
  description: string;
  willLearn: string[];
  willDo: string[];
}

File: backend/app/schemas/job.py
from marshmallow import Schema, fields


class JobSchema(Schema):
    id = fields.Str()
    icon = fields.Str()
    titleAr = fields.Str(attribute="title_ar")
    titleEn = fields.Str(attribute="title_en")
    descriptionPrimary = fields.Str(attribute="description_primary")
    descriptionSecondary = fields.Str(attribute="description_secondary")
    skills = fields.List(fields.Str())


class TaskSchema(Schema):
    id = fields.Int()
    type = fields.Str()
    title = fields.Str()
    fullTitle = fields.Str(attribute="full_title")
    duration = fields.Str()
    timeRange = fields.Str(attribute="time_range")
    description = fields.Str()
    willLearn = fields.List(fields.Str(), attribute="will_learn")
    willDo = fields.List(fields.Str(), attribute="will_do")


class JobWithTasksSchema(Schema):
    id = fields.Str()
    icon = fields.Str()
    titleAr = fields.Str(attribute="title_ar")
    titleEn = fields.Str(attribute="title_en")
    descriptionPrimary = fields.Str(attribute="description_primary")
    descriptionSecondary = fields.Str(attribute="description_secondary")
    skills = fields.List(fields.Str())
    tasks = fields.List(fields.Nested(TaskSchema))
